### PR TITLE
F#src 1632 add script save last ros logs

### PR DESCRIPTION
**Pull request by giusebar**

### CODE REVIEW

File: docker/utils/save_latest_ros_logs.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+if [ ! -z "$container_name" ]; then
+        container_array=($container_name)
+        for container in ${!container_array[@]}; do
+            current_container_name=${container_array[$container]}
+            echo "Copying logs from $current_container_name.."
+            ros_log_dir=~/Desktop/ROS_LOGS/$current_container_name
+            dir=ros_logs_$(date +%Y-%m-%d)
+            timestamp=$(date +%Y-%m-%d-%T)
+	        latestws=$(docker exec $current_container_name bash -c 'ls -dtr /home/user/.ros/wsdiff_ws_diff* | tail -1')
+            latestparam=$(docker exec agile_grasper_kinetic_real_hw bash -c 'ls -dtr /home/user/.ros/run_params* | tail -1')
```
Comment (giusebar) [inline docker/utils/save_latest_ros_logs.sh]: here I think there should be $current container name variable?

File: docker/utils/save_latest_ros_logs.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	        latestbag=$(docker exec agile_grasper_kinetic_real_hw bash -c 'ls -dtr /home/user/.ros/*.bag | tail -1')
```
Comment (giusebar) [inline docker/utils/save_latest_ros_logs.sh]: Same here

File: docker/utils/save_latest_ros_logs.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+            mkdir -p ${ros_log_dir}
+            mkdir -p ${ros_log_dir}/$dir
+            docker cp -L $current_container_name:home/user/.ros/log/latest ${ros_log_dir}/$dir
+
+            mv ${ros_log_dir}/$dir/latest ${ros_log_dir}/$dir/ros_log_$timestamp
+
+         	docker cp  -L $current_container_name:$latestws ${ros_log_dir}/$dir/ros_log_$timestamp
```
Comment (giusebar) [inline docker/utils/save_latest_ros_logs.sh]: Maybe you can add a folder here like:  ${ros_log_dir}/$dir/ros_log_$timestamp/latestws and so on, so it's easier to search

File: docker/utils/save_latest_ros_logs.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	        docker cp  -L $current_container_name:$latestparam ${ros_log_dir}/$dir/ros_log_$timestamp
```
Comment (giusebar) [inline docker/utils/save_latest_ros_logs.sh]: ${ros_log_dir}/$dir/ros_log_$timestamp/latestparam

File: docker/utils/save_latest_ros_logs.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    	    docker cp  -L $current_container_name:$latestbag ${ros_log_dir}/$dir/ros_log_$timestamp
```
Comment (giusebar) [inline docker/utils/save_latest_ros_logs.sh]: ${ros_log_dir}/$dir/ros_log_$timestamp/latestbag

File: docker/utils/save_latest_ros_logs.sh (hunk 3 above)
Comment (biofotis) [reply docker/utils/save_latest_ros_logs.sh]: its only one file so does it make sense to create a folder for a single file? That goes for the next 2 as well.

File: docker/launch.sh [hunk 6]
```diff
@@ -288,7 +289,8 @@ function optoforce_setup
 
 # If running for the first time create desktop shortcut
 APP_FOLDER=/home/$USER/.shadow_launcher_app
-BUILD_TOOLS_BRANCH=master
+SAVE_LOGS_APP_FOLDER=/home/$USER/.shadow_save_log_app
+BUILD_TOOLS_BRANCH=F%23SRC-1632_add_script_save_last_ros_logs
```
Comment (AndriyPt) [inline docker/launch.sh]: This need to be changed before merge

**Pull request merged**